fuzzbench/test_e2e_run.py

```diff
@@ -41,3 +45,12 @@ def test_db_contains_experiment_results(self):
     def test_experiment_report_is_generated(self):
         """Fake test to be implemented later."""
         assert True
+
+    def test_default_config_is_used(self):
+        """Tests whether the default config is used for end to end test."""
+        try:
+            fake_job = Job.fetch('e2e-test',
+                                 connection=redis.Redis(host='queue-server'))
+        except rq.exceptions.NoSuchJobError:
+            assert False
+        assert fake_job is not None
```

Comment:
Not sure what's the purpose of this check. The purpose of the e2e test is to do an as realistic as possible experiment run then check the result with these test_ functions. E.g. build a few images, then check here if the images were built successfully. 

If you'd like to check that current fake_jobs.bulid_image() jobs were successful, you can do that, eg under `test_all_jobs_finished_sucessfully()`.

Reply:
> Not sure what's the purpose of this check.

**This check is to see whether this end-to-end test use the correct/desired config file for experiment running.** The point lies that, we may hardcode some image name or jobs name and properties ahead in the test functions. If we are not certain this end-to-end test using our desired config or not, we are not sure the correctness to write other test functions.

> The purpose of the e2e test is to do an as realistic as possible experiment run then check the result with these test_ functions.

Yes! That is why we want to use a specified config file for an experiment run, other than making up the environment in the test class itself. Maybe `test_default_config_is_used` is not related to the real experiment running, however, it is a required step and give us confidence to write other end-to-end test. wdyt?

>However this check (test_default_config_is_used) doesn't really test "real code", but the "test code" that you added under the "if end-to-end test" branch.

The implementation is ugly, I admit. But the point is to let the test function to test whether we are using the desired test yaml.

> We can check that the current fake_jobs.bulid_image() jobs were successful on the other hand. Consider adding that to test_all_jobs_finished_sucessfully().

Good suggestion to look through.
By the way, I doubt whether we really need `test_all_jobs_finished_successfully` test. Should all jobs finish at last? This is more like an optimized desired behavior, other than a logic requirement/specification. Consider the coverage building task status in the future. We may want to pause those trial runners if some coverage build jobs fail.

Instead, I guess the test in #548 makes more sense since we assert the dependency logic for jobs working.